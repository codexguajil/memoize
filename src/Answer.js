import React, { Component } from 'react';

class Answer extends Component {
  constructor(props) {
    super(props)
  }

  checkAnswer = (e) => {
    console.log(e.target.innerHTML)
    this.props.checkAnswer(e.target.innerHTML);
  }

  render() {
    return (
      <div className="Answer" onClick={this.checkAnswer}>
        {this.props.answer}
      </div>
    )
  }
} 

export default Answer;