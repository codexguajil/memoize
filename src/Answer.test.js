import React from 'react';
import Answer from './Answer';
import { shallow, mount } from 'enzyme';

const mockAnswer = {
  "id": 11,
    "question": "Which command will show you the specific changes in any unstaged code?",
    "correctAnswer": "git diff",
    "allAnswers": [
      "git changes",
      "show diff",
      "git diff",
      "show change"
    ],
    "category": "git"
}

const toastLaunchMock = jest.fn();

describe("Answer", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
        <Answer answer={mockAnswer}
                correct={mockAnswer.correctAnswer}
                toastLaunch={toastLaunchMock}
        />
    )
  })

  it("should render one of the answer choices", () => {

    expect(wrapper).toMatchSnapshot()
  })

  it("should invoke checkAnswer on click of answer choice", () => {

    const e = {
      target: `<button className="Answer" onClick={this.checkAnswer}>
        git diff
      </button>`
    }

    wrapper.find('.Answer').simulate('click', e);
    
    
  })

}) 