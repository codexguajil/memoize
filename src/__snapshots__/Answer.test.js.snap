// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should render one of the answer choices 1`] = `
<div
  className="Answer"
  onClick={[Function]}
  value={
    Object {
      "allAnswers": Array [
        "git changes",
        "show diff",
        "git diff",
        "show change",
      ],
      "category": "git",
      "correctAnswer": "git diff",
      "id": 11,
      "question": "Which command will show you the specific changes in any unstaged code?",
    }
  }
>
  <Component />
</div>
`;
